### PR TITLE
added authentication and authorization in whole app

### PR DESCRIPTION
**Pull request by jstsanam**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -41,8 +41,10 @@ passport.deserializeUser(User.deserializeUser());     // user related info sessi
 // Implementing flash messages
 app.use(flash());
 app.use((req, res, next) => {
+  //locals ase variables jo hr jgah accessible h
```
Comment (sanchit203) [inline app.js]: is this a whatsapp chat??

File: middlewares/isAuthorized.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import Listing from "../schemas/listingSchema.js";
+import Review from "../schemas/reviewSchema.js";
+
+const isListingAuthor = async (req, res, next) => {
```
Comment (sanchit203) [inline middlewares/isAuthorized.js]: name of this function is not intutive.. the name should `isUserListingOwner`

File: middlewares/isAuthorized.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    //Checking if user is actual owner of listing or not
+    if (!listing.owner.equals(res.locals.currUser._id)) {
+        req.flash("error", "Only owners can update the listing 🙅🏻‍♀️");
+        return res.redirect(`/listings/${id}`);
+    }
+    next();
+}
+
+const isReviewAuthor = async (req, res, next) => {
```
Comment (sanchit203) [inline middlewares/isAuthorized.js]: same here change the name

File: middlewares/isAuthorized.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    let { id, reviewId } = req.params;
+    let review = await Review.findById(reviewId);
+
+    //Checking if user is actual author of review or not
```
Comment (sanchit203) [inline middlewares/isAuthorized.js]: always have a space after the // eg. `// Checking if user is actual author of review or not`

**Review by sanchit203 — DISMISSED** [review]

File: app.js (hunk 1 above)
Comment (jstsanam) [reply app.js]: done

File: middlewares/isAuthorized.js (hunk 2 above)
Comment (jstsanam) [reply middlewares/isAuthorized.js]: done

File: middlewares/isAuthorized.js (hunk 3 above)
Comment (jstsanam) [reply middlewares/isAuthorized.js]: done

**Review by sanchit203 — APPROVED** [review]

**Pull request merged**